godot-core/src/obj/dyn_gd.rs

```diff
@@ -478,3 +480,44 @@ where
     D: ?Sized + 'static,
 {
 }
+
+impl<T, D> Var for DynGd<T, D>
+where
+    T: GodotClass,
+    D: ?Sized + 'static,
+{
+    fn get_property(&self) -> Self::Via {
+        self.obj.get_property()
+    }
+
+    fn set_property(&mut self, value: Self::Via) {
+        // `set_property` can't be delegated to Gd<T> since we have to set `erased_obj` as well
+        *self = <Self as FromGodot>::from_godot(value);
+    }
+}
+
+impl<T, D> Export for DynGd<T, D>
+where
+    T: GodotClass + Bounds<Exportable = bounds::Yes>,
+    D: ?Sized + 'static,
+{
+    fn export_hint() -> PropertyHintInfo {
+        let hint: PropertyHint;
+        if T::inherits::<classes::Node>() {
+            hint = PropertyHint::NODE_TYPE;
+        } else if T::inherits::<classes::Resource>() {
+            hint = PropertyHint::RESOURCE_TYPE;
+        } else {
+            // theoretically unreachable - `Object`s are not suitable for export.
+            return PropertyHintInfo {
+                hint: PropertyHint::NONE,
+                hint_string: GString::default(),
+            };
```

Comment:
i think it'd be better to just use `unreachable!` here, that's what we do with gd: https://github.com/godot-rust/gdext/blob/6a5d19d5f8e7563730ccd4738e46f398726b222a/godot-core/src/obj/gd.rs#L919

maybe honestly it'd be better to just defer to the `Gd` impl but change the hint string? like 
```rs
PropertyHintInfo {
  hint_string: get_dyn_property_hint_string::<D>(),
  ..Gd::<T>::export_hint(),
}
```

Reply:
Deffering to Gd<T> seems like obvious choice :sweat_smile:, implemented